Bottom: bdb9d273b2d50f133425bec9b2b8f977424bb435
Top:    17cf277efcd7fd372c6e659a3205d3fe9ce571cb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-11 15:31:22 +0000

Workaround crash where contour is passed as NULL to poly_PreContour













---

diff --git a/src/polygon.c b/src/polygon.c
index 1999b43..471b91c 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -253,6 +253,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
